src/main/clm_varpar.F90

```diff
@@ -227,20 +228,31 @@ subroutine clm_varpar_init(actual_maxsoil_patches, actual_numcft)
        write(iulog, *)
     end if
 
+    ! TODO BFB refactor: How to derive these instead of hardwiring?
```

Comment:
I gave this a try today and failed due to the order of initializations in the code:

Deriving `ndecomp_pools` and `ndecomp_cascade_transitions` on the fly rather than hardwiring, means initializing them in `init_decompcascade_bgc` or `init_decompcascade_cn` or 'init_decompcascade_mimics`. However, we use these parameters earlier in `InitAllocate` (in `SoilBiogeochemStateType`).

Reordering initializations feels like a bigger refactor than I should attempt here.

Reply:
This is fine with me, @slevisconsulting .  Is it helpful to have another meeting, or do you feel like there's more you can continue working on?

Reply:
As @ekluzek recommended, I created a refactor branch and am working on the refactor ideas that have come up this far. Soon thereafter we should meet. Would Tuesday or Monday next week work? I'm happy to book it if I can point to the same zoom info as we used last time. Otherwise feel free to book it @wwieder because my zoom acct permits up to 40-min meetings.